linkerd/buffer/src/service.rs

```diff
@@ -61,14 +47,13 @@ impl<Req, F> Buffer<Req, F> {
     }
 }
 
-impl<Req, F> tower::Service<Req> for Buffer<Req, F>
+impl<Req, Rsp> tower::Service<Req> for Buffer<Req, Rsp>
 where
-    F: TryFuture,
-    F::Error: Into<Error>,
+    Rsp: Send + 'static,
 {
-    type Response = F::Ok;
+    type Response = Rsp;
     type Error = Error;
-    type Future = ResponseFuture<F>;
+    type Future = Pin<Box<dyn Future<Output = Result<Rsp, Error>> + Send + 'static>>;
```

Comment:
i'd be curious to see if switching to nightly and returning `impl Future` resulted in the same build time improvement or not --- but we can do that separately. it would be good to know eventually, since that could help guide future changes...